src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp

```diff
@@ -450,6 +459,14 @@ void S3ObjectStorage::copyObjectMultipartImpl(
 
         throwIfError(outcome);
     }
+
+    if (settings_ptr->s3_settings.check_objects_after_upload)
+    {
+        auto object_head = requestObjectHeadData(dst_bucket, dst_key);
```

Comment:
Does it make retries?

Reply:
Yes, retries are fully controlled by AWS client.